src/cargo/core/compiler/job_queue.rs

```diff
@@ -839,29 +850,29 @@ impl<'cfg> DrainState<'cfg> {
         }
 
         let time_elapsed = util::elapsed(cx.bcx.config.creation_time().elapsed());
-        if let Err(e) = self.timings.finished(cx, &error) {
-            if error.is_some() {
-                crate::display_error(&e, &mut cx.bcx.config.shell());
-            } else {
+        if let Err(e) = self.timings.finished(cx, &errors) {
+            if errors.is_success() {
                 return Some(e);
+            } else {
+                crate::display_error(&e, &mut cx.bcx.config.shell());
```

Comment:
I realize that this was preexisting, but reading over this again I'm finding it a bit confusing to follow why above cases use `handle_error` but `self.timings.finish` and the `writeln!` below both use diferent custom logic. I think, however, that it should be fine to use `handle_error` in both cases?

Or rather put another way I think it'd be good to consistently use `handle_error` here if possible.